### PR TITLE
add asciidoctor-diagrams support 

### LINKED ISSUE #76: Detect diagram type
I use kramdoc as a converter to be able to transform markdown to pdf (Asciidoc is awesome, but our documentation generator only support markdown).

My current script looks like this:

```sh
#!/usr/bin/env sh
#This scripts require the installation of the ruby gems asciidoctor, asciidoctor-pdf, kramdoc and rouge

kramdoc "$1" -o - | asciidoctor-pdf -a "source-highlighter=rouge" -r rouge -o "${1%.*}.pdf" -
```

I would love to be able to also generate diagrams during the process, using asciidoctor-diagram.

Howerver, if i write in my markdown file :

\`\`\`plantuml
@startuml
Alice --> Bob: Hello
@enduml
\`\`\`

It will be converted to

```asciidoc
[source, plantuml]
----
@startuml
Alice --> Bob: Hello
@enduml
----
```

when this output would be more useful.

```asciidoc
[plantuml]
----
@startuml
Alice --> Bob: Hello
@enduml
----
```

### PR DESCRIPTION
**Pull request by pomdtr**
Closes #76 

Please be indulgent, I have never coded in ruby.

### CODE REVIEW

**Comment by pomdtr** [comment]
Maybe it's more right to convert

\`\`\`plantuml
Alice --> bob
\`\`\`

to

```asciidoc
[plantuml]
....
Alice --> Bob
....
```

**Comment by mojavelinux** [comment]
Yes, it should be on a literal block, not a listing block.

File: lib/kramdown-asciidoc/converter.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# encoding: UTF-8
```
Comment (mojavelinux) [inline lib/kramdown-asciidoc/converter.rb]: Please don't add the encoding magic comment. That's no longer needed in Ruby.

**Comment by mojavelinux** [comment]
Looking good so far!

**Comment by pomdtr** [comment]
I don't have any experience in ruby, so this syntax error is kind of cryptic to me ¯\_(ツ)_/¯

**Comment by pomdtr** [comment]
Ok, it seem to work now. Should i add an unit test ?

**Comment by mojavelinux** [comment]
Yes. Though they aren't really unit tests. They are document tests. You put the input and output and the test suite will verify they match. You can use one of the existing document tests as a reference.

**Comment by mojavelinux** [comment]
What concerns me a bit about this is the proliferation of diagram names we have to track. That seems like it is going to be a maintenance problem. Is it possible to narrow this down to less names, or is there a place we can get this list of names? Surely Markdown isn't supporting all the same diagram names as Asciidoctor Diagram. Or is it?

**Comment by pomdtr** [comment]
Hi, feel free to update the pull requests with the necessary changes.

I do not need this feature anymore, and I don't have much time to resolve the merge conflicts.

Have a great day !

**Comment by mojavelinux** [comment]
Thanks for letting me know. I'll take it from here. Even though I haven't gotten around to merging it yet, thanks for submitting the PR!

**Comment by mojavelinux** [comment]
Replace by #105.

**Pull request closed without merging**